### ISSUE TITLE
SslStream ShutdownAsync for Net 4.0 and line based protocol session handing.

### ISSUE THREAD

**Issue by nvth198z** [chronoxor/NetCoreServer#39]
Hi chronoxor,
Your TcpServer not using pre-buffer allocate strategy but there is no memory leak, it's a good server, but it using .Net default ThreadPool, it's cannot handle a lot of session, I think so.

I'm trying to compile your library (just some class) with .Net framework 4.0, but in SslSession and SslServer require ShutdownAsync to work, is there any alternative method to shutdown SslSession and SslServer asynchronous in .Net 4.0 ?

I want to develop a line based protocol program (like SMTP, POP3...) is there any idea for read line by line from your TcpSession class ?

**Comment by chronoxor** [comment]
The server don't use `Thread per Session` scheme. It use ThradPool only to handle async events. It is very close to 'one working thread per core' scheme to handle all sessions.

**Comment by nvth198z** [comment]
Yes, I know you not using thread per session, but if I change to SmartThreadPool, the process is not consume a lot of memory, I think the problem is ThreadPool itself
